tests/unit/core/test_Wallet.py

```diff
@@ -0,0 +1,136 @@
+from twisted.trial import unittest
+
+from twisted.internet import threads, defer
+from lbrynet.core.Wallet import Wallet
+
+test_metadata = {
+'license': 'NASA',
+'fee': {'USD': {'amount': 0.01, 'address': 'baBYSK7CqGSn5KrEmNmmQwAhBSFgo6v47z'}},
+'ver': '0.0.3',
+'description': 'test',
+'language': 'en',
+'author': 'test',
+'title': 'test',
+'sources': {
+    'lbry_sd_hash': '8655f713819344980a9a0d67b198344e2c462c90f813e86f0c63789ab0868031f25c54d0bb31af6658e997e2041806eb'},
+'nsfw': False,
+'content_type': 'video/mp4',
+'thumbnail': 'test'
+}
+
+
+class MocLbryumWallet(Wallet):
+    def __init__(self):
+        pass
+    def get_name_claims(self):
+        return threads.deferToThread(lambda: [])
+
+    def _save_name_metadata(self, name, claim_outpoint, sd_hash):
+        return defer.succeed(True)
+
+
+class WalletTest(unittest.TestCase):
+
+    def _check_exception(self, d):
+        def check(err):
+            with self.assertRaises(Exception):
+                err.raiseException()
+        d.addCallbacks(lambda _: self.assertTrue(False), lambda err: check(err))
+
+    def test_failed_send_name_claim(self):
+        def not_enough_funds_send_name_claim(self, name, val, amount):
+            claim_out = {'success':False, 'reason':'Not enough funds'}
+            return claim_out
+        MocLbryumWallet._send_name_claim = not_enough_funds_send_name_claim
+        wallet = MocLbryumWallet()
+        d = wallet.claim_name('test', 1, test_metadata)
+        self._check_exception(d)
+        return d
+
+    def test_successful_send_name_claim(self):
+        test_claim_out = {
+            "claimid": "f43dc06256a69988bdbea09a58c80493ba15dcfa",
+            "fee": "0.00012",
+            "nout": 0,
+            "success": True,
+            "txid": "6f8180002ef4d21f5b09ca7d9648a54d213c666daf8639dc283e2fd47450269e"
+         }
+
+        def check_out(claim_out):
+            self.assertTrue('success' not in claim_out)
+            self.assertEqual(claim_out['claimid'], test_claim_out['claimid'])
+            self.assertEqual(claim_out['fee'], test_claim_out['fee'])
+            self.assertEqual(claim_out['nout'], test_claim_out['nout'])
+            self.assertEqual(claim_out['txid'], test_claim_out['txid'])
```

Comment:
for some reason, I don't know why, it is convention to put the expected part of the test first:

`self.assertEqual(expected_claim_out['txid'], actual_claim_out['txid'])`